ciao-cli/configuration.go

```diff
@@ -0,0 +1,261 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/01org/ciao/ciao-controller/api"
+	"github.com/01org/ciao/ciao-controller/types"
+)
+
+var configCommand = &command{
+	SubCommands: map[string]subCommand{
+		"update": new(configUpdateCommand),
+		"show":   new(configShowCommand),
+	},
+}
+
+type configUpdateCommand struct {
+	Flag    flag.FlagSet
+	element string
+	value   string
+}
+
+func (cmd *configUpdateCommand) usage(...string) {
+	fmt.Fprintf(os.Stderr, `
+Usage:
+
+ciao-cli [options] config update [flags]
+
+update cluster configuration
+
+The update flags are:
+
+`)
+	cmd.Flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func (cmd *configUpdateCommand) parseArgs(args []string) []string {
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.StringVar(&cmd.element, "element", "", "configuration field to change")
+	cmd.Flag.StringVar(&cmd.value, "value", "", "new value for element to change")
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+func (cmd *configUpdateCommand) run(args []string) error {
+	var req types.ConfigRequest
+	var response interface{}
+	if cmd.element == "" {
+		fmt.Fprintf(os.Stderr, "Missing required -element parameter\n")
+		cmd.usage()
+	}
+	if cmd.value == "" {
+		fmt.Fprintf(os.Stderr, "Missing required -value parameter\n")
+		cmd.usage()
+	}
+	if validConfigElement(cmd.element) == false {
+		fmt.Fprintf(os.Stderr, "'%s' is not a valid element of the cluster configuration\n", cmd.element)
+		os.Exit(2)
+	}
+	if validConfigValue(cmd.value, cmd.element) == false {
```

Comment:
You could define new types for element and value and implement the Set(val string) error method.  This way cmd.flag.Parse will do all the checking for you.  See https://github.com/01org/ciao/blob/master/ciao-launcher/tests/ciaolc/ciaolc.go#L66